docs(v1.8.0): add CHANGELOG (#10264)
longhorn/longhorn: CHANGELOG/CHANGELOG-1.8.0.md

## Longhorn v1.8.0 Release Notes

This latest version of Longhorn introduces several features, enhancements, and bug fixes that are intended to improve system quality and the overall user experience. Highlights include new V2 Data Engine features, multiple backupstores, RWX Volume, installation, and OS Supportability, and stability, performance, and resilience.

The Longhorn team appreciates your contributions and anticipates receiving feedback regarding this release.

For more information about release-related terminology, see [Releases](https://github.com/longhorn/longhorn#releases).

> [!IMPORTANT]

> Due to the upgrade of the CSI external snapshotter to version v8.2.0, ensure that all clusters are running Kubernetes v1.25 or later before upgrading to Longhorn v1.8.0 or any newer version.

Reviewer: ```suggestion
The CSI external-snapshotter was upgraded to v8.2.0. Ensure that all clusters are running Kubernetes v1.25 or later before upgrading to Longhorn v1.8.0 or a later version.
```

## Deprecation & Incompatibilities

> - The default block size for block-type disks was 4096 bytes prior to v1.8.0. However, a 512-byte block size is more commonly used and aligns with the v1 data engine's configuration. Additionally, the 4096-byte block size is incompatible with backing images generated by the v1 data engine. To address this, the default block size has been changed to 512 bytes. For existing v2 volumes, users can update their setup by following these steps:

Reviewer: ```suggestion
- The default block size for block-type disks in earlier Longhorn releases is 4096 bytes. However, 512 bytes is more commonly used and aligns with the V1 Data Engine's configuration. Additionally, the 4096-byte block size is incompatible with backing images generated by the V1 Data Engine. To address these concerns, the default block size was changed to 512 bytes.
```

>     - Back up the current v2 volumes.
>     - Remove the v2 volumes.
>     - Delete the block-type disk with a 4096-byte block size from node.spec.disks.
>     - Erase the old data on the block-type disk using tools such as dd.
>     - Re-add the disk to node.spec.disks with the updated configuration.
>     - Restore the v2 volumes.

Reviewer: ```suggestion
  If you have existing V2 volumes, perform the following steps:

    1. Back up the V2 volumes.
    2. Remove the V2 volumes.
    3. Delete the block-type disk with a 4096-byte block size from `node.spec.disks`.
    4. Erase the old data on the block-type disk using tools such as dd.
    5. Add the disk again to `node.spec.disks` with the updated configuration.
    6. Restore the V2 volumes.
```

    For more information, see [#10053](https://github.com/longhorn/longhorn/issues/10053).

> - A data corruption in v2 volume that affects earlier Longhorn releases has been resolved in v1.8.0. The issue involves potential continual changes to the checksum of files in a V2 volume with multiple replicas. This occurs because SPDK allocates clusters without initialization, leading to data inconsistencies across replicas. The varying data read from the volume can result in data corruption and broken backups. For more information, see [#10053](https://github.com/longhorn/longhorn/issues/10135).

Reviewer: ```suggestion
- A V2 volume data corruption issue that affects earlier Longhorn releases has been resolved in v1.8.0. The issue involves potential continual changes to the checksum of files in a V2 volume with multiple replicas. This occurs because SPDK allocates clusters without initialization, leading to data inconsistencies across replicas. The varying data read from the volume can result in data corruption and broken backups. For more information, see [#10053](https://github.com/longhorn/longhorn/issues/10135).
```

## Primary Highlights

### New V2 Data Engine Features

Although the V2 Data Engine is still considered an experimental feature in this release, the core functions have been significantly enhanced.

> - [configurable CPU cores](https://longhorn.io/docs/1.8.0/v2-data-engine/features/configurable-cpu-cores): Support the global and node-specific configuration options provide greater control and flexibility for optimizing performance and resource allocation.

Reviewer 2: ```suggestion
- [Configurable CPU cores](https://longhorn.io/docs/1.8.0/v2-data-engine/features/configurable-cpu-cores): Support the global and node-specific configuration options provide greater control and flexibility for optimizing performance and resource allocation.
```

Author: Fixed. Thank you.

- [Disaster recovery (DR) volumes](https://github.com/longhorn/longhorn/issues/6613): Designed to store data in a backup cluster in the event of a failure in the main cluster. DR volumes enhance the resiliency of Longhorn volumes by ensuring data can be quickly restored in case of cluster outages.
- [Auto-salvage volumes](https://github.com/longhorn/longhorn/issues/8430): Automatically repair volumes in the event of a failure.
- [Live migration](https://github.com/longhorn/longhorn/issues/6361): Allow for the migration of volumes from one node to another without interrupting the services using those volumes.

> - [Volume encryption](https://github.com/longhorn/longhorn/issues/7355): Ensure data stored in volumes is protected through encryption at rest. 

Reviewer 2: ```suggestion
- [Volume encryption](https://github.com/longhorn/longhorn/issues/7355): Ensure data stored in volumes is protected through encryption in transit and at rest. 
```

Author: Fixed. Thank you.

- [Delta replica rebuilding using snapshot checksum](https://github.com/longhorn/longhorn/issues/9488).
- [Backing image update and download](https://github.com/longhorn/longhorn/issues/6341).

> ### Backupstore
> 
> Starting with v1.8.0, Longhorn introduces support for [multiple backupstores](https://github.com/longhorn/longhorn/issues/5411), enhancing earlier versions that were limited to a single backup target for accessing a backupstore.

Reviewer: ```suggestion
### Multiple Backupstores and Default Backup Target

Starting with v1.8.0, Longhorn allows you to use multiple backupstores for storing backups of Longhorn volumes. Longhorn v1.8.0 also creates a default backup target (`default`) during installation and upgrades. The default backup target is used for system backups and volumes that were created without an assigned backup target name.

[Demo](https://www.youtube.com/watch?v=cvsz6YLDLrM) | [Documentation](https://longhorn.io/docs/1.8.0/snapshots-and-backups/backup-and-restore/set-backup-target/) | [GitHub Issue](https://github.com/longhorn/longhorn/issues/5411)
```

> ### RWX Volume
> 
> Longhorn supports fully [automatic online expansion of RWX volumes](https://longhorn.io/docs/1.8.0/nodes-and-volumes/volumes/expansion/#rwx-volume). There is no need to scale down the workload or apply manual commands. Full details are in RWX Volume

Reviewer: ```suggestion
### Automatic RWX Volume Expansion

Longhorn v1.8.0 supports fully automatic online expansion of RWX volumes without the need to scale down the workload or apply manual commands. To use this feature, ensure that the v1.8.0 versions of Longhorn Manager, Share Manager, and the CSI plugin are all running.

[Documentation](https://longhorn.io/docs/1.8.0/nodes-and-volumes/volumes/expansion/#rwx-volume) | [GitHub Issue](https://github.com/longhorn/longhorn/issues/9736)
```

### Installation

> Longhorn also supports installation or upgrade via the HelmChart controller built into RKE2 and K3s. It allows management in a CRD YAML chart of most of the options that would normally be passed to the helm command-line tool. For more details on how it works, see [Install with Helm Controller](https://longhorn.io/docs/1.8.0/nodes-and-volumes/volumes/expansion/#rwx-volume).

Reviewer 2: The link is wrong.

Author: Fixed. Thank you.

> ### Installation
> 
> Longhorn also supports installation or upgrade via the HelmChart controller built into RKE2 and K3s. It allows management in a CRD YAML chart of most of the options that would normally be passed to the helm command-line tool. For more details on how it works, see [Install with Helm Controller](https://longhorn.io/docs/1.8.0/deploy/install/install-with-helm-controller).

Reviewer: ```suggestion
### Helm Controller

You can now install and upgrade Longhorn on clusters running RKE2 or K3s using the Helm Controller that is built into those distributions. The Helm Controller manages Helm charts using a HelmChart Custom Resource Definition (CRD), which contains most of the options that would normally be passed to the Helm command-line tool.

[Documentation](https://longhorn.io/docs/1.8.0/deploy/install/install-with-helm-controller/) | [GitHub Issue](https://github.com/longhorn/longhorn/issues/9506)
```

> ### OS Supportability
> 
> To use V2 volumes in Talos Linux clusters, ensure that all nodes meet the V2 Data Engine prerequisites. For more information, see [Talos Linux Support: V2 Data Engine](https://longhorn.io/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support#v2-data-engine).

Reviewer: ```suggestion
### V2 Data Engine Support for Talos Linux

Longhorn v1.8.0 supports usage of V2 volumes in Talos Linux clusters. To use this feature, ensure that all nodes meet the V2 Data Engine prerequisites. 

[Documentation](https://longhorn.io/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support#v2-data-engine) | [GitHub Issue](https://github.com/longhorn/longhorn/issues/7791)
```

For more information about release-related terminology, see [Releases](https://github.com/longhorn/longhorn#releases).

> [!IMPORTANT]
The CSI external-snapshotter was upgraded to v8.2.0. Ensure that all clusters are running Kubernetes v1.25 or later before upgrading to Longhorn v1.8.0 or a later version.

## Deprecation & Incompatibilities

- The default block size for block-type disks in earlier Longhorn releases is 4096 bytes. However, 512 bytes is more commonly used and aligns with the V1 Data Engine's configuration. Additionally, the 4096-byte block size is incompatible with backing images generated by the V1 Data Engine. To address these concerns, the default block size was changed to 512 bytes.

  If you have existing V2 volumes, perform the following steps:

    1. Back up the V2 volumes.
    2. Remove the V2 volumes.
    3. Delete the block-type disk with a 4096-byte block size from `node.spec.disks`.
    4. Erase the old data on the block-type disk using tools such as dd.
    5. Add the disk again to `node.spec.disks` with the updated configuration.
    6. Restore the V2 volumes.

    For more information, see [#10053](https://github.com/longhorn/longhorn/issues/10053).

> - A V2 volume data corruption issue that affects earlier Longhorn releases has been resolved in v1.8.0. The issue involves potential continual changes to the checksum of files in a V2 volume with multiple replicas. This occurs because SPDK allocates clusters without initialization, leading to data inconsistencies across replicas. The varying data read from the volume can result in data corruption and broken backups. For more information, see [#10053](https://github.com/longhorn/longhorn/issues/10135).

Reviewer 2: ```suggestion
- A V2 volume data corruption issue that affects earlier Longhorn releases has been resolved in v1.8.0. The issue involves potential continual changes to the checksum of files in a V2 volume with multiple replicas. This occurs because SPDK allocates clusters without initialization, leading to data inconsistencies across replicas. The varying data read from the volume can result in data corruption and broken backups. For more information, see [#10035](https://github.com/longhorn/longhorn/issues/10135).
```
